test/integration/deployment/deployment_test.go

```diff
@@ -69,3 +73,76 @@ func TestNewDeployment(t *testing.T) {
+	updatedDeploymentExtensionsV1beta1, err := c.ExtensionsV1beta1().Deployments(ns.Name).Update(deploymentExtensionsV1beta1)
+	if err != nil {
+		t.Fatalf("failed to update extensions/v1beta1 deployment %s: %v", deploymentExtensionsV1beta1.Name, err)
+	}
+	if !reflect.DeepEqual(updatedDeploymentExtensionsV1beta1.Spec.Selector.MatchLabels, newSelectorLabels) {
+		t.Errorf("selector should be changed for extensions/v1beta1, expected: %v, got: %v", newSelectorLabels, updatedDeploymentExtensionsV1beta1.Spec.Selector.MatchLabels)
+	}
+
+	// test to ensure apps/v1beta1 selector is mutable
+	deploymentAppsV1beta1, err := c.AppsV1beta1().Deployments(ns.Name).Get(updatedDeploymentExtensionsV1beta1.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get apps/v1beta1 deployment %s: %v", updatedDeploymentExtensionsV1beta1.Name, err)
+	}
+
+	newSelectorLabels = map[string]string{"name_apps_v1beta1": "test_apps_v1beta1"}
+	deploymentAppsV1beta1.Spec.Selector.MatchLabels = newSelectorLabels
+	deploymentAppsV1beta1.Spec.Template.Labels = newSelectorLabels
+	updatedDeploymentAppsV1beta1, err := c.AppsV1beta1().Deployments(ns.Name).Update(deploymentAppsV1beta1)
+	if err != nil {
+		t.Fatalf("failed to update apps/v1beta1 deployment %s: %v", deploymentAppsV1beta1.Name, err)
+	}
+	if !reflect.DeepEqual(updatedDeploymentAppsV1beta1.Spec.Selector.MatchLabels, newSelectorLabels) {
+		t.Errorf("selector should be changed for apps/v1beta1, expected: %v, got: %v", newSelectorLabels, updatedDeploymentAppsV1beta1.Spec.Selector.MatchLabels)
+	}
+
+	// test to ensure apps/v1beta2 selector is immutable
+	// TODO(#50791): change this to expect an error after selector immutability codes are moved to validation
+	deploymentAppsV1beta2, err := c.AppsV1beta2().Deployments(ns.Name).Get(updatedDeploymentAppsV1beta1.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get apps/v1beta2 deployment %s: %v", updatedDeploymentAppsV1beta1.Name, err)
+	}
+	newSelectorLabels = map[string]string{"name_apps_v1beta2": "test_apps_v1beta2"}
+	deploymentAppsV1beta2.Spec.Selector.MatchLabels = newSelectorLabels
+	deploymentAppsV1beta2.Spec.Template.Labels = newSelectorLabels
+	_, err = c.AppsV1beta2().Deployments(ns.Name).Update(deploymentAppsV1beta2)
+	if err == nil {
+		t.Fatalf("failed to provide validation error when changing immutable selector when updating apps/v1beta2 deployment %s", deploymentAppsV1beta2.Name)
+	}
+	expectedErrList := field.ErrorList{
+		&field.Error{
+			Type:  field.ErrorTypeInvalid,
+			Field: field.NewPath("spec").Child("selector").String(),
+			BadValue: &metav1.LabelSelector{
+				MatchLabels: newSelectorLabels,
+			},
+			Detail: "selector must not be changed after update",
+		},
+	}
+	expectedErrMessage := fmt.Sprintf("Deployment.apps \"%s\" is invalid: %v", name, expectedErrList.ToAggregate())
+	errMessage := fmt.Sprintf("%v", err)
+	if errMessage != expectedErrMessage {
```

Comment:
You can just do `if err.Error() != ...`.

Reply:
Thanks, this is more concise!